pkg/query/flamegraph_arrow.go

```diff
@@ -194,6 +198,43 @@ func generateFlamegraphArrowRecord(ctx context.Context, mem memory.Allocator, tr
 				fb.parent.Set(rootRow)
 				row = fb.builderCumulative.Len()
 			}
+			if fb.aggregationConfig.aggregateByTimestamp {
+				tsHasher.Reset()
+				tsStr := strconv.FormatInt(r.Timestamp.Value(i), 10)
+				durationStr := strconv.FormatInt(r.Duration.Value(i), 10)
+				_, _ = tsHasher.Write([]byte(tsStr))
+				_, _ = tsHasher.Write([]byte(durationStr))
+				tsHash = tsHasher.Sum64()
+
+				sampleTsRow := row
+				if row, ok := fb.rootsRow[tsHash]; ok {
```

Comment:
for flamecharts having a root for a timestamp that already exists means we can't possibly build a correct flamechart, we should error on this case

Reply:
I'm sorry I don't understand why this has to be errored. 

So what I'm trying to here is convert the following records
```
[{
    ts: timestamp-1
    samples: ....
  },
  {
    ts: timestamp-2
    samples: ....
  },
  {
    ts: timestamp-1
    samples: .....
}]
```
to 
```
[{
    ts:timestamp-1
    samples: ....
  },
  {
    ts: tiemstamp-2
    samples: ....
}]
```

So that we'll have all samples with the same timestamp are grouped under a single root, that is easier for rendering in the UI. 
Isn't this similar to the group by label here:
<img width="570" alt="Screenshot 2024-12-16 at 3 03 30 PM" src="https://github.com/user-attachments/assets/17b5d683-60c4-4620-86bc-6ac3648c2feb" />

Reply:
There can't ever be two samples at the same timestamp for the same thread or CPU. Merging should happen when `end timestamp == next timestamp` (within some error of margin because clocks aren't perfectly precise, maybe 1-10 ms).

Reply:
And `end timestamp = timestamp + value`